### ISSUE TITLE
Server Crash when a player leaves a party

### ISSUE THREAD

**Issue by TomRotfl** [nwnxee/unified#920]
This is weird, and very random. I don't know if one or more players are doing this intentionally or not, it started happening a few weeks ago, while still on 8193.5 and it's still a thing on 8193.13.

When this happens, every player online start to see "Player has left the party" spammed on server log chat, even if that player had nothing to do with these players. Right after that, server is crashing.

I couldn't reproduce or anything.

I am not hooked to anything about party, NWNX side, and there is nothing custom for when a player leaves a party in my module.

I attach a screenshot of other players seeing the Server log chat spam, right before the crash and server crash report.

![partycrash](https://user-images.githubusercontent.com/60175349/82763478-633b9e80-9e08-11ea-8824-c1d98bfbade0.png)

[nwserver-crash-1590331292.log](https://github.com/nwnxee/unified/files/4674290/nwserver-crash-1590331292.log)

Thank you

**Comment by mtijanic** [comment]
@TomRotfl if this happens again after the latest updates to nwnx (as of today), please attach new logs. It should have more info to debug, because right now it doesn't.

**Comment by plenarius** [comment]
@TomRotfl Does this still happen for you?

**Comment by TomRotfl** [comment]
Yes, it happened today as well (planning to update to .16 in the next few hours tho, so i am still at .13 with latest .13 nwnx release)

![image](https://user-images.githubusercontent.com/60175349/95079792-667c5a80-0717-11eb-9add-512c9951fa8f.png)

**Comment by Morderon** [comment]
Place the below in NWNX_ON_PARTY_LEAVE_BEFORE event. Create it if you don't currently have it subscribed.

`object oSelf = OBJECT_SELF;
    object oPC   = GetFirstFactionMember(oSelf);

    int nPC;
    while(GetIsObjectValid(oPC))
    {
        nPC++;
        if(nPC > 1)
	    break;
        oPC = GetNextFactionMember(oSelf);
    }

    if(nPC == 1)
        NWNX_Events_SkipEvent();`

**Comment by TomRotfl** [comment]
I will give this a try. I was already checking something similar, but only for `NWNX_ON_PARTY_KICK_BEFORE`, but looks like it wasn't working. I'll test this one and let you know. Thanks

**Comment by plenarius** [comment]
Been trying to look more into this,. Do you do any other things related to faction? `NWNX_Creature_SetFaction` or `NWNX_Encounter_SetFactionId` for example? Same question to you too @FrogsOfDoom

Don't understand how the event bypass described by Morderon could be effective while nwn is crashing or in an infinite loop.

**Comment by TomRotfl** [comment]
For me, hooking `NWNX_ON_PARTY_LEAVE_BEFORE` seems to have done the trick (fingers crossed)

I am not using neither of those functions in my mod.

**Comment by FrogsOfDoom** [comment]
> Been trying to look more into this,. Do you do any other things related to faction? `NWNX_Creature_SetFaction` or `NWNX_Encounter_SetFactionId` for example? Same question to you too @FrogsOfDoom

Not using either of those functions. It's a one-off on our end but going to try adding the hotfix in anyway; can't do harm.
